org/mozilla/jss/netscape/security/x509/AlgorithmId.java

```diff
@@ -552,7 +714,64 @@ public final boolean equals(ObjectIdentifier id) {
         return algid.equals(id);
     }
 
+    public static AlgorithmParameters createPSSAlgorithmParametersFromData(byte[] der) throws Exception {
+        if (der == null) {
+            throw new Exception("Invalid input data.");
+        }
+        AlgorithmParameters pssParams = null;
+        try {
+            pssParams = AlgorithmParameters.getInstance("RSAPSSAlgorithmParameters", "Mozilla-JSS");
+        } catch (NoSuchProviderException e) {
+            throw new Exception(e);
+        }
 
+        try {
+            pssParams.init(der);
+        } catch (IOException e) {
+            throw new Exception("Error intializing RSAPSS parameters: " + e);
+        }
+        return pssParams;
+    }
+    /* Used to create the PSS algorithm params needed for RSA PSS signatures
+     * for now only support the RSAPSS algoritm with the SHA256 digest
+     * this can be extended to support SHA384 and 512 PSS digests.
+    */
+    public static AlgorithmParameters createPSSAlgorithmParameters(String algName) throws Exception {
+        if (algName == null) {
+            throw new Exception("Invalid Algorithm name input.");
+        }
+        AlgorithmParameters pssParams = null;
+        PSSParameterSpec pssSpec = null;
+        // Make sure we are in the RSA PSS family
+        if (algName.contains("PSS")) {
```

Comment:
This'd be simpler if we wrote:

`if (!algName.contains("PSS")) { throw Exception(...) }` and then we wouldn't need the body indented.

Reply:
Done.